error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/CollectionRules.java

```diff
@@ -35,13 +35,21 @@ private CollectionRules() {}
    */
   static final class CollectionIsEmpty<T> {
     @BeforeTemplate
-    @SuppressWarnings("java:S1155" /* This violation will be rewritten. */)
+    @SuppressWarnings({
+      "java:S1155" /* This violation will be rewritten. */,
+      "LexicographicalAnnotationAttributeListing" /* `key-*` entry must remain last. */,
+      "OptionalFirstCollectionElement" /* This is a more specific template. */,
+      "StreamIsEmpty" /* This is a more specific template. */,
+      "key-to-resolve-AnnotationUseStyle-and-TrailingComment-check-conflict"
+    })
     boolean before(Collection<T> collection) {
       return Refaster.anyOf(
           collection.size() == 0,
           collection.size() <= 0,
           collection.size() < 1,
-          Iterables.isEmpty(collection));
+          Iterables.isEmpty(collection),
+          collection.stream().findAny().isEmpty(),
+          collection.stream().findFirst().isEmpty());
```

Comment:
So `OptionalFirstCollectionElement` and `StreamIsEmpty` prefer different variants. But our Refaster integration prefers writing the longest expression matched, so this one will win.

Reply:
Understood, I was wondering indeed what happens when an expression is matched by multiple rules. It makes sense that the most specific one wins 👍

Reply:
Yeah; implemented [here](https://github.com/PicnicSupermarket/error-prone-support/blob/1cc792c615ff9a3fdb164506255dda7e78069ad8/refaster-runner/src/main/java/tech/picnic/errorprone/refaster/runner/Refaster.java#L100-L131) :)